### PR TITLE
clean up alloc and version check

### PR DESCRIPTION
**Pull request by Sasasu**
and fix memory leak maybe
and improve performance

### CODE REVIEW

**Review by xuebinsu — APPROVED** [review]

File: diskquota_utility.c [hunk 1]
```diff
@@ -1087,56 +1125,72 @@ set_per_segment_quota(PG_FUNCTION_ARGS)
...
-	        "(t.quotaType = %d OR t.quotaType = %d) AND t.primaryOid = q.targetOid AND t.quotaType = q.quotaType",
-	        spcoid, NAMESPACE_TABLESPACE_QUOTA, ROLE_TABLESPACE_QUOTA);
-
-	ret = SPI_execute(buf.data, true, 0);
+	ret = SPI_execute_with_args(
+	        "SELECT true FROM diskquota.target AS t, diskquota.quota_config AS q WHERE tablespaceOid = $1 AND "
+	        "(t.quotaType = $2 OR t.quotaType = $3) AND t.primaryOid = q.targetOid AND t.quotaType = q.quotaType",
+	        3,
+	        (Oid[]){
+	                INT4OID,
```
Comment (beeender) [inline diskquota_utility.c:1152]: should be `OIDOID`

File: diskquota_utility.c [hunk 2]
```diff
@@ -1199,43 +1253,6 @@ worker_spi_get_extension_version(int *major, int *minor)
 	return ret;
 }
 
-/*
- * Get major version from extversion, and convert it to int
- * 0 means an invalid major version.
- */
-int
-get_ext_major_version(void)
```
Comment (beeender) [inline diskquota_utility.c:1207]: :clap:

File: diskquota_utility.c [hunk 3]
```diff
@@ -1087,56 +1125,72 @@ set_per_segment_quota(PG_FUNCTION_ARGS)
...
+	        "SELECT true FROM diskquota.target AS t, diskquota.quota_config AS q WHERE tablespaceOid = $1 AND "
+	        "(t.quotaType = $2 OR t.quotaType = $3) AND t.primaryOid = q.targetOid AND t.quotaType = q.quotaType",
+	        3,
+	        (Oid[]){
+	                INT4OID,
+	                INT4OID,
+	                INT4OID,
+	        },
+	        (Datum[]){
+	                Int32GetDatum(spcoid),
```
Comment (beeender) [inline diskquota_utility.c:1157]: `ObjectIDGet`

File: quotamodel.c [hunk 4]
```diff
@@ -983,24 +982,16 @@ flush_to_table_size(void)
 
 	initStringInfo(&insert_statement);
 	appendStringInfo(&insert_statement, "insert into diskquota.table_size values ");
```
Comment (beeender) [inline quotamodel.c:984]: not related with this PR, but indention looks funny here
image

**Review by beeender — COMMENTED** [review]
And please mention why do we remove the `get_ext_major_version` in the commit message -- the new way of supporting upgrade will ensure the old DDL won't run with new so file.

File: quotamodel.c (hunk 4 above)
Comment (Sasasu) [reply quotamodel.c:984]: I think the newly expanded part does not format with `./.editorconfig`.

GitHub's bug

**Review by beeender — APPROVED** [review]

**Pull request merged**